### PR TITLE
Created test directory and create test cases for post functionality

### PR DESCRIPTION
**Pull request by haiimkeith**
## Description / Changes Made
- Create unit_tests directory to work with run_tests.py
- Added tests cases for server.py functions including get_post(), create_post(), update_post(), and delete_post()

## How to Test

1. Run backend API server by using python3 server.py -m http.server
2. On a separate terminal run python3 run_tests.py
3. Check if tests pass or fail, then also check coverage report.

## Checklist

- [x] I have added/updated relevant documentation, and I have followed the coding style guidelines.
- [X] I have added/updated tests, and I have run the test suite and all tests pass.
- [X] I have checked for any potential conflicts with other branches and fixed any merge conflicts.

### CODE REVIEW

File: backend/unit_tests/test_update_post.py
Comment (noodaj) [inline backend/unit_tests/test_update_post.py:1]: You should also check for what happens when the post fails to update. What do we do then? Do we have a response that is sent back so we can determine what happens next in our frontend code?

**Review by noodaj — CHANGES_REQUESTED** [review]
For all of these test cases are we not able to combine them into one big test file. It doesn't make sense to a majority of the code for each of the files are very similar.
- Setting up base and user url
- Connect to the db
-  Creating the new user test data

I think that these first 3 steps can be put at the very top of the **SINGLE** file that we have and every test can be written similar to what you have as a function underneath the **setup**  steps. This will help us test different edge cases instead of having to update every file to test that edge case. 

You should also check for what happens each of the functions **FAILS** to do its job. What do we do then? Do we have a response that is sent back so we can determine what happens next in our frontend code?

**Review by honeepuffs — CHANGES_REQUESTED** [review]
I found your individual files well organized. This organization allowed me to easily understand each of these tests, however upon reading Jadon's comment, I agree. 
I think possibly adding well organized comment headers of what you are testing would be beneficial. So the file although larger may still be organized and be able to be searched through easily.

**Review by honeepuffs — APPROVED** [review]
Your clear documentation and combination of the files is great! You were able to fufill both requests and I see no issues as to why we cannot pull request. Great job!

**Pull request merged**